etna/models/nn/timesfm.py

```diff
@@ -0,0 +1,362 @@
+        """
+        return self
+
+    def predict(
+        self,
+        ts: TSDataset,
+        prediction_size: int,
+        return_components: bool = False,
+    ) -> TSDataset:
+        """Make predictions using true values as autoregression context (teacher forcing).
+
+        Parameters
+        ----------
+        ts:
+            Dataset with features.
+        prediction_size:
+            Number of last timestamps to leave after making prediction.
+            Previous timestamps will be used as a context.
+        return_components:
+            If True additionally returns forecast components.
+
+        Returns
+        -------
+        :
+            Dataset with predictions.
+        """
+        raise NotImplementedError("Method predict isn't currently implemented!")
+
+    def _get_exog_features(self) -> List[str]:
+        static_reals = [] if self.static_reals is None else self.static_reals
+        static_categoricals = [] if self.static_categoricals is None else self.static_categoricals
+        time_varying_reals = [] if self.time_varying_reals is None else self.time_varying_reals
+        time_varying_categoricals = [] if self.time_varying_categoricals is None else self.time_varying_categoricals
+
+        return static_reals + static_categoricals + time_varying_reals + time_varying_categoricals
+
+    def forecast(
+        self,
+        ts: TSDataset,
+        prediction_size: int,
+        return_components: bool = False,
+    ) -> TSDataset:
+        """Make autoregressive forecasts.
+
+        Parameters
+        ----------
+        ts:
+            Dataset with features.
+        prediction_size:
+            Number of last timestamps to leave after making prediction.
+            Previous timestamps will be used as a context.
+        return_components:
+            If True additionally returns forecast components.
+
+        Returns
+        -------
+        :
+            Dataset with predictions.
+
+        Raises
+        ------
+        NotImplementedError:
+            if return_components mode is used.
+        ValueError:
+            if dataset doesn't have any context timestamps.
+        ValueError:
+            if there are NaNs in the middle or end of the time series.
+        NotImplementedError:
+            if forecasting is done without exogenous features and dataset has None frequency.
+        """
+        if return_components:
+            raise NotImplementedError("This mode isn't currently implemented!")
+
+        max_context_size = len(ts.index) - prediction_size
+        if max_context_size <= 0:
+            raise ValueError("Dataset doesn't have any context timestamps.")
+
+        if max_context_size < self.context_size:
+            warnings.warn("Actual length of a dataset is less that context size. All history will be used as context.")
+
+        self.tfm._set_horizon(prediction_size)
+
+        end_idx = len(ts.index)
+
+        all_exog = self._get_exog_features()
```

Comment:
We could name it like `exog_columns` to highlight that this is a list of columns.